### PR TITLE
ECS guns

### PR DESCRIPTION
**Pull request by metalgearsloth**
After this I want to implement my refactor to cleanup gun code and then after that do prediction.

I removed requires because gh wasn't having a great time.

### CODE REVIEW

**Comment by metalgearsloth** [comment]
Merge after https://github.com/space-wizards/space-station-14/pull/6033 because fixing that PR is probably more CBT than this one.

**Comment by ElectroJr** [comment]
> Merge after #6033 because fixing that PR is probably more CBT than this one.

Just looking at the files changed I don't think they would even conflict? And either way, comparatively the other one is a much smaller PR than this one, so it should be easier to fix. So if anything this should be merged first.

Also review wise, I should hopefully have time to go through this sometime this weekend.

**Comment by metalgearsloth** [comment]
It should be identical to the non-ECS version with the main changes being:
- The GridCoordinates-y firing message just uses EntityCoordinates
- C# events removed

File: Content.Server/Content.Server.csproj [hunk 1]
```diff
@@ -25,6 +25,9 @@
     <ProjectReference Include="..\RobustToolbox\Robust.Server\Robust.Server.csproj" />
     <ProjectReference Include="..\Content.Shared\Content.Shared.csproj" />
   </ItemGroup>
+  <ItemGroup>
+    <Compile Remove="Weapon\Ranged\Barrels\BarrelSystem.cs" />
+  </ItemGroup>
```
[the remark this answers is not in the document]
Comment (metalgearsloth) [reply Content.Server/Content.Server.csproj:30]: AHHHHHHHHHH RIDER

**Pull request merged**